### PR TITLE
Remove a bunch of old gamemode code

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Hardly tested, will be test merging it live.

Removes a bunch of unused old gamemode code, as well as refactoring the important bits outside of gamemodes. Prepares it for when `/datum/game_mode` dies.

Do not merge until:
1. Testing, which also can't occur until...
2. https://github.com/tgstation/tgstation13.org/pull/67 is merged, which removes the mode dependency for the front page. I need to confirm with MSO that everything is prepared.

### CODE REVIEW

**Comment by Farquaar** [comment]
Improve don't remove :^)

**Comment by Mothblocks** [comment]
mso still hasnt merged pr so im going to break the website

**Comment by Mothblocks** [comment]
image

annoying fucking case insensitive names

**Comment by Mothblocks** [comment]
merge #59220 first i'm not dealing with this

**Comment by Mothblocks** [comment]
Test merged on Sybil where it says the server is down LIKE I WAAAAARNED

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -661,18 +621,6 @@ SUBSYSTEM_DEF(ticker)
 			//Break chain since this has a sleep input in it
 			addtimer(CALLBACK(player, /mob/dead/new_player.proc/make_me_an_observer), 1)
 
-/datum/controller/subsystem/ticker/proc/load_mode()
```
Comment (AnturK) [inline code/controllers/subsystem/ticker.dm:664]: Is there some persistent method for configuring dynamic next round ? Events used to be set up one round before then resetting the server. Also helps with local dev to keep it same.

File: code/datums/world_topic.dm [hunk 2]
```diff
@@ -141,7 +141,6 @@
 /datum/world_topic/status/Run(list/input)
 	. = list()
 	.["version"] = GLOB.game_version
-	.["mode"] = GLOB.master_mode
```
Comment (AnturK) [inline code/datums/world_topic.dm:144]: I assume banner code and such were changed ?

File: code/modules/events/_event.dm [hunk 3]
```diff
@@ -23,9 +23,6 @@
 	var/alert_observers = TRUE //should we let the ghosts and admins know this event is firing
 									//should be disabled on events that fire a lot
 
-	var/list/gamemode_blacklist = list() // Event won't happen in these gamemodes
```
Comment (AnturK) [inline code/modules/events/_event.dm:26]: Removing these without replacement doesn't stil well with me but i suppose we're running dynamic long enough that if these were an issue someone would complain.

File: code/modules/station_goals/dna_vault.dm [hunk 4]
```diff
@@ -11,6 +11,9 @@
 #define VAULT_SPEED "Leg Muscle Stimulus"
 #define VAULT_QUICK "Arm Muscle Stimulus"
 
+/// List of available station goals for the crew to be working on
+GLOBAL_LIST_EMPTY_TYPED(station_goals, /datum/station_goal)
```
Comment (AnturK) [inline code/modules/station_goals/dna_vault.dm:15]: Shouldn't this be in some other file ?

**Review by AnturK — COMMENTED** [review]
Looks good just few questions.

File: code/datums/world_topic.dm (hunk 2 above)
Comment (Mothblocks) [reply code/datums/world_topic.dm:144]: After I complained about it for 2 weeks, yes 😄

Comment (Mothblocks) [reply code/datums/world_topic.dm:144]: This was TM'd and Sybil reported as down, but then it got fixed.

File: code/modules/station_goals/dna_vault.dm (hunk 4 above)
Comment (Mothblocks) [reply code/modules/station_goals/dna_vault.dm:15]: Oops, yes, didn't notice where it was.

File: code/controllers/subsystem/ticker.dm (hunk 1 above)
Comment (Mothblocks) [reply code/controllers/subsystem/ticker.dm:664]: Persistent? No, but you can force the roundstart ruleset to be extended without anyone seeing.

**Pull request merged**